src/prover.rs

```diff
@@ -0,0 +1,101 @@
+mod settings;
+
+use crate::args::ProveArgs;
+use crate::utils::write_json_to_file;
+use settings::{get_default_prover_config, get_default_prover_parameters};
+use std::path::PathBuf;
+use std::process::Command;
+use stone_prover_sdk::error::ProverError;
+use stone_prover_sdk::json::read_json_from_file;
+
+/// Runs the Stone prover with the given inputs
+///
+/// # Arguments
+///
+/// * `air_public_input` - Path to the public input file
+/// * `air_private_input` - Path to the private input file
+/// * `tmp_dir` - Temporary directory for intermediate files
+///
+/// # Returns
+///
+/// A `Result` containing `StoneProverResult` on success, or an `Error` on failure
+pub fn run_stone_prover(
+    prove_args: &ProveArgs,
+    air_public_input: &PathBuf,
+    air_private_input: &PathBuf,
+    tmp_dir: &tempfile::TempDir,
+) -> Result<(), ProverError> {
+    println!("Running prover...");
+
+    let tmp_prover_parameters_path = tmp_dir.path().join("prover_parameters.json");
+
+    let prover_parameters_path = if let Some(parameter_file) = &prove_args.parameter_file {
+        parameter_file
+    } else {
+        let air_public_input_json: serde_json::Value =
```

Comment:
I know this is Moonsongs code, but any reason they cannot just deserialize the entire JSON object into a struct using serde?

Reply:
Good point. Changed.